common/types/explorer.ts

```diff
@@ -174,12 +174,13 @@ export interface SavedVisualization extends SavedObjectAttributes {
   selected_fields: { text: string; tokens: [] };
   selected_timestamp: IField;
   type: string;
-  subType?: 'metric' | 'visualization' | typeof PROMQL_METRIC_SUBTYPE; // exists if sub type is metric
+  subType?: 'metric' | 'visualization'; // exists if sub type is metric
   user_configs?: string;
   units_of_measure?: string;
   application_id?: string;
   dataSources: string; // list of type SelectedDataSources that is stringified
   queryLang: string;
+  metricType?: typeof PROMQL_METRIC_SUBTYPE | typeof OTEL_METRIC_SUBTYPE; // exists if sub type is metric
```

Comment:
why not merge with `subType`? if an older cluster has `subType = 'promqlmetric'`, would the code still work since the code now expects `metricType = 'promqlmetric'`

Reply:
The older cluster would have the `subType` as `metric` as the `promqlmetric` type was introduced to `subType` in the Peter's recent PR merged couple of weeks ago.  
MetricType is created in an assumption that more metrics types can be introduced in future and this variable would be only one required to differentiate between all metric type. SubType is to differentiate between a visualization and metric